src/components/data/DataTable.js

```diff
@@ -497,6 +501,17 @@ const DataTable = props => {
       },
       onDismiss: () => setUpdatingEntity(undefined)
     }),
+    !!renamingColumn && h(RenameColumnModal, {
+      namespace, name,
+      entityType: _.head(entities).entityType,
+      oldAttributeName: renamingColumn,
+      onSuccess: () => {
+        setRenamingColumn(undefined)
+        Ajax().Metrics.captureEvent(Events.workspaceDataRenameColumn, extractWorkspaceDetails(workspace.workspace))
+        loadMetadata()
```

Comment:
Not necessary to change, but FYI the already loaded entityMetadata could be updated using `setEntityMetadata` to avoid this request.